### PR TITLE
rpmsgfs related update

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

rpmsgfs related update
1. add wait ready to rpmsgfs
2. add timeout when mount in rpmsgfs
3. fix blocked when connecting
4. use rptun_wait/post to resolve deadlock with usrsock rpmsg

This depends on:
https://github.com/apache/incubator-nuttx/pull/5847/

## Impact

rpmsgfs

## Testing

VELA

### CODE REVIEW

File: fs/rpmsgfs/rpmsgfs.c [hunk 1]
```diff
@@ -252,6 +259,21 @@ static void rpmsgfs_mkpath(FAR struct rpmsgfs_mountpt_s  *fs,
...
+      struct stat buf;
+      int ret;
+
+      ret = rpmsgfs_client_stat(fs->handle, fs->fs_root, &buf);
+      if (ret == 0)
+        {
+          break;
+        }
+
+      usleep(RPMSGFS_RETRY_DELAY_MS * 1000);
```
Comment (pkarashchenko) [inline fs/rpmsgfs/rpmsgfs.c:274]:
```suggestion
      usleep(RPMSGFS_RETRY_DELAY_MS * USEC_PER_MSEC);
```

File: fs/rpmsgfs/rpmsgfs.c [hunk 2]
```diff
@@ -1101,6 +1129,7 @@ static int rpmsgfs_bind(FAR struct inode *blkdriver, FAR const void *data,
    */
 
   fs->fs_head = NULL;
+  fs->timeout = INT32_MAX;
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs.c:1131]: INT32_MAX->INT_MAX

File: fs/rpmsgfs/rpmsgfs.c [hunk 3]
```diff
@@ -265,7 +265,7 @@ static void rpmsgfs_mkpath(FAR struct rpmsgfs_mountpt_s *fs,
       struct stat buf;
       int ret;
 
-      ret = rpmsgfs_client_stat(fs->handle, path, &buf);
+      ret = rpmsgfs_client_stat(fs->handle, fs->fs_root, &buf);
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs.c:268]: merge to previous patch

File: fs/rpmsgfs/rpmsgfs.c [hunk 4]
```diff
@@ -1103,7 +1103,6 @@ static int rpmsgfs_bind(FAR struct inode *blkdriver, FAR const void *data,
           fs->timeout = atoi(&ptr[8]);
         }
 
-
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs.c:1078]: move to previous patch

File: fs/rpmsgfs/rpmsgfs.c [hunk 5]
```diff
@@ -252,6 +259,21 @@ static void rpmsgfs_mkpath(FAR struct rpmsgfs_mountpt_s  *fs,
...
+      int ret;
+
+      ret = rpmsgfs_client_stat(fs->handle, fs->fs_root, &buf);
+      if (ret == 0)
+        {
+          break;
+        }
+
+      usleep(RPMSGFS_RETRY_DELAY_MS * USEC_PER_MSEC);
+      fs->timeout -= RPMSGFS_RETRY_DELAY_MS;
```
Comment (pkarashchenko) [inline fs/rpmsgfs/rpmsgfs.c:275]: This is not very good for delay calculation. I mean that the error is very big due to dependency on systick granularity. The better way is to use `clock_systime_timespec()`.

Comment (GUIDINGLI) [reply fs/rpmsgfs/rpmsgfs.c:275]: We don't need so accurate time, just a timeout

**Review by xiaoxiang781216 — APPROVED** [review]

File: fs/rpmsgfs/rpmsgfs.c [hunk 6]
```diff
@@ -252,6 +259,21 @@ static void rpmsgfs_mkpath(FAR struct rpmsgfs_mountpt_s  *fs,
     {
       strncat(path, &relpath[first], pathlen - strlen(path) - 1);
     }
+
+  while (fs->timeout > 0)
```
Comment (pkarashchenko) [inline fs/rpmsgfs/rpmsgfs.c:263]: Does `INT_MAX` timeout means "wait forever"?

Comment (xiaoxiang781216) [reply fs/rpmsgfs/rpmsgfs.c:263]: Almost, 0x7fffffff/1000/60/60/24=24.5days.

Comment (pkarashchenko) [reply fs/rpmsgfs/rpmsgfs.c:263]: Haha... The question is do we support infinite wait somehow or any value is considered a timeout?

Comment (pkarashchenko) [reply fs/rpmsgfs/rpmsgfs.c:263]: The current code handles any positive value as a timeout, so no support of infinite wait. I just want to make sure that it is by design

Comment (GUIDINGLI) [reply fs/rpmsgfs/rpmsgfs.c:263]: Yes.
That I found the default value setting in wrong place, so update it.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**